client/blocks/site/index.jsx

```diff
@@ -102,7 +125,7 @@ class Site extends React.Component {
 							  } )
 							: site.domain
 					}
-					onClick={ this.onSelect }
+					onClick={ this.onViewSiteClick }
```

Comment:
The `Site` component is reused at a plenty of places and overriding the `onClick` handler breaks many of them. The `onSelect` prop should not be ignored. On `homeLink` should trigger the view behavior:
```js
onClick={ this.homeLink ? this.onViewSiteClick : this.onSelect }
```

Test by going to `/plans` and choosing a site from the site selector. Click should perform the select and navigate to the site's Plans page. At this moment, it navigates to `/view/site.blog` 👎

Reply:
Nice catch! Fixed in: https://github.com/Automattic/wp-calypso/commit/e3e2154f145f98f5277836f0394c31319f05b0f8